metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/AbstractMetisPlugin.java

```diff
@@ -176,10 +178,20 @@ public PluginStatus getPluginStatus() {
   }
 
   /**
+   * This method also clears the fail message if the status is set to anything other than {@link
+   * PluginStatus#FAILED}.
+   *
    * @param pluginStatus {@link PluginStatus}
    */
   public void setPluginStatus(PluginStatus pluginStatus) {
     this.pluginStatus = pluginStatus;
+    if (this.pluginStatus != PluginStatus.FAILED) {
```

Comment:
I think the failMessage should always be reset not only for non-FAILED statuses.
Also probably better to have it in a different call e.g. setPluginStatusAndResetFailMessage so that it's clear what the end result is, and not using the default setter.

Reply:
Agreed. i changed it.